src/lib/checksums.py

```diff
@@ -0,0 +1,74 @@
+import hashlib
+import logging
+import typing as t
+
+from .errors import SourceUpdateError
+
+
+log = logging.getLogger(__name__)
+
+
+class MultiDigest(t.NamedTuple):
+    md5: t.Optional[str] = None
+    sha1: t.Optional[str] = None
+    sha256: t.Optional[str] = None
+    sha512: t.Optional[str] = None
+
+    @classmethod
+    def form_source(cls, source: t.Dict):
```

Comment:
Sorry to nitpick but should this be `from_source`? and `-> MultiDigest`?

Reply:
Right. And the code autocomplete made sure that the typo got successfully propagated everywhere.